services/server/src/server/apiv2/errors.ts

```diff
@@ -0,0 +1,23 @@
+import { NotFoundError } from "../../common/errors";
+import { v4 as uuidv4 } from "uuid";
+
+export type ErrorCode = "unsupported_chain";
+
+export interface GenericErrorResponse {
+  customCode: ErrorCode;
+  message: string;
+  errorId: string;
+}
+
+export class ChainNotFoundError extends NotFoundError {
+  payload: GenericErrorResponse;
+
+  constructor(message: string) {
+    super(message);
+    this.payload = {
+      customCode: "unsupported_chain",
```

Comment:
I would specify a list of all the custom codes that we have. Not a complete one, but let's initialize it

Reply:
I already created an issue for that: https://github.com/ethereum/sourcify/issues/1855
I would rather keep it in the scope of another PR

Reply:
To clarify, the list should go into the `ErrorCode` type